compiler/z/codegen/S390Peephole.cpp

```diff
@@ -1349,29 +1349,89 @@ TR_S390PostRAPeephole::ConditionalBranchReduction(TR::InstOpCode::Mnemonic branc
 
 
 /**
- * Catch the pattern where an CR/BRC can be converted
- *    CR R1,R2
- *    BRC  Mask, Lable
+ * Catch the pattern where an CLR/BRC can be converted
+ *    CLR R1, R2
+ *    BRC Mask, Lable
  * Can be replaced with
- *    CRJ  R1,R2,Lable
+ *    CLRJ R1, R2, Lable, Mask
  */
 bool
 TR_S390PostRAPeephole::CompareAndBranchReduction()
    {
-   bool performed = false;
+   if (!TR::Compiler->target.cpu.getSupportsArch(TR::CPU::z10))
+      return false;
 
-   TR::Instruction *current = _cursor;
-   TR::Instruction *next = _cursor->getNext();
-   TR::InstOpCode::Mnemonic curOpCode = current->getOpCodeValue();
-   TR::InstOpCode::Mnemonic nextOpCode = next->getOpCodeValue();
-   if ((curOpCode == TR::InstOpCode::CR || curOpCode == TR::InstOpCode::CGR || curOpCode == TR::InstOpCode::CGFR)
-      && (nextOpCode == TR::InstOpCode::BRC || nextOpCode == TR::InstOpCode::BRCL))
-      {
-      printf("Finding CR + BRC\n");
-      printf("method=%s\n", comp()->signature());
+   bool branchTakenPerformReduction = false;
+   bool fallThroughPerformReduction = false;
+   TR::Instruction *clrInstruction = _cursor;
+   TR::Instruction *brcInstruction = _cursor->getNext();
+   TR::Instruction *clrjInstruction = NULL;
+   TR::LabelSymbol *labelSymbol = NULL;
+
+   if (clrInstruction->getOpCodeValue() == TR::InstOpCode::CLR
+       && brcInstruction->getOpCodeValue() == TR::InstOpCode::BRC)
+      {
+      /* if branch target is a snippet we'll go ahead with the
+       * reduction. Otherwise, we look at next 5 (or until TR::BBEnd
+       * depending on whichever comes first) to check if any
+       * instruction reads or sets the condition code.
+       * We'll perform the reduction if an instruction sets
+       * the condition code or, if none of the instructions read the
+       * condition code set by clrInstruction
+       */
+      labelSymbol = brcInstruction->getLabelSymbol();
+      if (labelSymbol->getSnippet())
+         {
+         branchTakenPerformReduction = true;
+         }
+      else
+         {
+            // check branch case for condition code usage
+            int branchTakenInstIndex = 0;
```

Comment:
Prefer to use explicitly typed variables (`int32_t` in this case), or `auto` to guarantee precise variable width for portability (`int` is defined as having a minimum of 4-bytes in width, but there is no maximum).